Assets/FallingPlatform.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FallingPlatform : MonoBehaviour
+{
+    // Start is called before the first frame update
+    [SerializeField]
+    GameObject spriteObject;
+    [SerializeField]
+
+    float elapsed = 0.0f;
+    [SerializeField]
+
+    float duration = 1.5f;
+    [SerializeField]
+
+    float shakeMagnitudeX = 0.0f;
+    [SerializeField]
+
+    float shakeMagnitudeY = 0.1f;
+    [SerializeField]
+
+    float RestoreTime = 5f;
+    Vector3 spriteOriginalPos;
+    Vector3 objectOriginalPos;
+    Rigidbody2D localRB;
+
+    void Start()
+    {
+
+        spriteOriginalPos = spriteObject.transform.position;
+        objectOriginalPos = transform.position;
+        localRB = GetComponent<Rigidbody2D>();
+
+    }
+    void Restore()
+    {
+        localRB.isKinematic = true;
+        localRB.constraints = RigidbodyConstraints2D.FreezeAll;
+        transform.position = objectOriginalPos;
+        spriteObject.transform.position = spriteOriginalPos;
+        elapsed = 0.0f;
+    }
+    IEnumerator OnCollisionEnter2D(Collision2D other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            StartCoroutine(Shake());
+            yield return new WaitForSeconds(duration);
+            Fall();
+            yield return new WaitForSeconds(RestoreTime);
+            Restore();
+        }
```

Comment:
I think we also need to prevent the coroutine from running again if the player jumps on the platform again before it falls - there are a few ways to do this.

Easiest way prob is to add an `isTriggered` bool or something like that:

```suggestion
        if (!isTriggered && other.gameObject.CompareTag("Player"))
        {
            isTriggered = true;
            StartCoroutine(Shake());
            yield return new WaitForSeconds(duration);
            Fall();
            yield return new WaitForSeconds(RestoreTime);
            Restore();
            isTriggered = false;
        }
```